File: EasyTab/EasyTab.Services/Database/Reservation.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace EasyTab.Services.Database;

public partial class Reservation
{
    public int Id { get; set; }

    public int UserId { get; set; }

    public int TableId { get; set; }

    public DateTime ReservationDate { get; set; }

    public TimeOnly StartTime { get; set; }

    public TimeOnly EndTime { get; set; }

    public DateTime CreatedAt { get; set; }

    public int? ApprovedById { get; set; }

    public DateTime? ApprovedAt { get; set; }

    public int? CancelledById { get; set; }

    public DateTime? CancelledAt { get; set; }

    public string? CancellationReason { get; set; }

    public virtual Table Table { get; set; } = null!;

    public virtual User User { get; set; } = null!;

    public virtual User? ApprovedBy { get; set; }

    public virtual User? CancelledBy { get; set; }

    [MaxLength(1000)]
    public string ReservationState { get; set; } = string.Empty;
}

File: EasyTab/UI/easytab_mobile/lib/models/reservation.dart
import 'package:json_annotation/json_annotation.dart';

part 'reservation.g.dart';

@JsonSerializable()
class Reservation {
  final int? id;
  final int? userId;
  final String? firstName;
  final String? lastName;
  final int? tableId;
  final String? tableName;
  final int? numberOfGuests;
  final DateTime? reservationDate;
  final String? startTime;
  final String? endTime;
  final DateTime? createdAt;
  final String? reservationState;
  final int? approvedById;
  final DateTime? approvedAt;
  final int? cancelledById;
  final DateTime? cancelledAt;
  final String? cancellationReason;
  final int? localeId;
  final String? localeName;
  final String? localeAddress;
  final String? localeLogo;

  Reservation({
    this.id,
    this.userId,
    this.firstName,
    this.lastName,
    this.tableId,
    this.tableName,
    this.numberOfGuests,
    this.reservationDate,
    this.startTime,
    this.endTime,
    this.createdAt,
    this.reservationState,
    this.approvedById,
    this.approvedAt,
    this.cancelledById,
    this.cancelledAt,
    this.cancellationReason,
    this.localeId,
    this.localeName,
    this.localeAddress,
    this.localeLogo,
  });

  String get customerName => '${firstName ?? ''} ${lastName ?? ''}'.trim();

  factory Reservation.fromJson(Map<String, dynamic> json) =>
      _$ReservationFromJson(json);
  Map<String, dynamic> toJson() => _$ReservationToJson(this);
}
